external-crates/move/move-compiler/src/typing/core.rs

```diff
@@ -84,6 +84,8 @@ pub struct Context<'env> {
 
     /// collects all called functions in the current module
     pub called_fns: BTreeSet<Symbol>,
+    /// collects all used constants in the current module
+    pub used_consts: BTreeSet<Symbol>,
```

Comment:
You should be able to walk the annotations and just pessimistically assume that `AttributeValue::ModuleAccess` is a constant. 
I think you will need to add the module ident, so `BTreeSet<ModuleIdent, Symbol>`, and then gather this information at the end of typing. It should work. 

To future proof, I would probably just do this for functions too (and private types) too. Since they all share the same name space, we could make this `used_module_members`

Reply:
cc @cgswords who will be doing something similar (or has already done something similar) in terms of gathering up information from modules

Reply:
Noted! And I am indeed changing this to use one collection as per @tnowacki's earlier suggestion.